### PR TITLE
Add ngscopeclient (scopehal-apps) oscilloscope software and FFTS fast fourier transform library as a package

### PR DESCRIPTION
**Pull request by hansfbaier**
## Description of changes
This packages the ngscopeclient app (project scopehal-apps)
https://www.ngscopeclient.org/
and its dependency:
This packages the FFTS fast fourier transform library:
https://github.com/anthonix/ffts

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by MaxwellDupre — APPROVED** [review]
LGTM

File: pkgs/development/libraries/ffts/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, cmake, git, lib, fetchFromGitHub,
+  ... }:
```
Comment (bgamari) [inline pkgs/development/libraries/ffts/default.nix:2]: The `...` isn't necessary.

File: pkgs/applications/science/electronics/scopehal-apps/default.nix [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    in
+      builtins.concatStringsSep ":" (map libAppend buildInputs);
+
+  cmakeFlags = [
+    "-DCURRENT_GIT_VERSION=${lib.substring 0 7 src.rev}"
+    "-Wno-deprecated"
+  ];
+
+  postInstall = ''
+    ln -s $out/share $out/bin/
```
Comment (bgamari) [inline pkgs/applications/science/electronics/scopehal-apps/default.nix:65]: There must be a better way.

File: pkgs/applications/science/electronics/scopehal-apps/default.nix [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    mv -v $out/bin/ngscopeclient $out/bin/ngscopeclient-unwrapped
+    makeWrapper $out/bin/ngscopeclient-unwrapped $out/bin/ngscopeclient --set LD_LIBRARY_PATH="${libraryPaths}"
```
Comment (bgamari) [inline pkgs/applications/science/electronics/scopehal-apps/default.nix:67]: The `LD_LIBRARY_PATH` bit shouldn't be necessary.

**Review by bgamari — CHANGES_REQUESTED** [review]
There are a few issues here. I will rebase and try to address them.

**Comment by bgamari** [comment]
I have opened #336646 with an updated and tested variant of this work.

File: pkgs/applications/science/electronics/scopehal-apps/default.nix (hunk 3 above)
Comment (hansfbaier) [reply pkgs/applications/science/electronics/scopehal-apps/default.nix:67]: If it isn't there ngscopeclient does not start up. Running nix on Arch that is.

**Pull request closed without merging**